dvc/remote/base.py

```diff
@@ -153,6 +153,12 @@ def group(self, name):
     def cache(self):
         return getattr(self.repo.cache, self.scheme)
 
+    def get_checksum_type_list(self):
```

Comment:
Let's define something like
DEFAULT_CHECKSUMS = ["md5"]
for each remote instead of PARAM_CHECKSUM and then use it if `checksums` are not specified in the configs. Then make something like
```
@property
def checksums(self):
    return self._checksums

@property
def checksum(self):
    return self._checksums[0]
```
and use that, so you don't have to define your getters for each remote.

Reply:
Isn't this same thing?

Reply:
but without this change, I still don't have to define getters for each remote either?
Isn't it inherit from remote base?

Reply:
Tried to define DEFAULT_CHECKSUM_TYPES but some tests like TestCacheExists are setting PARAM_CHECKSUM after init

Which is against from what you said, don't know how it got accepted in the code base 🙂 

>Overwriting constants is not nice.

Reply:
Well, yeah, you'll have to adjust those to do things things properly.

It is acceptable for tests and unacceptable to overwrite constants in the core code itself.

Reply:
> It is acceptable for tests and unacceptable to overwrite constants in the core code itself.

IMHO, it is a bad decision.
Unacceptable things should be keeping consistent for both the core code and tests. They are under the same project, right? Find excuse to write unacceptable code in the project is not nice. 🙂 
Otherwise, means the tests are using the core code in the unacceptable ways.
Hmmm, maybe it is because you are the committer.🙂